File: MošPosudit.Services/DataBase/Data/Review.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MošPosudit.Services.DataBase.Data
{
    public class Review
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        public int ToolId { get; set; }

        [Required]
        [Range(1, 5)]
        public int Rating { get; set; }

        [Required]
        [StringLength(1000)]
        public string? Comment { get; set; }

        [Required]
        public DateTime CreatedAt { get; set; }

        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        [ForeignKey("UserId")]
        public User User { get; set; }
        [ForeignKey("ToolId")]
        public Tool Tool { get; set; }

        // Logging properties
        [NotMapped]
        public string EntityName => "Review";

        [NotMapped]
        public string DisplayName => $"Review by {User?.Username} for {Tool?.Name}";
    }
} 

File: MosPosudit.UI/shared/lib/models/review.dart
class ReviewModel {
  final int id;
  final int userId;
  final String? userName;
  final int toolId;
  final String? toolName;
  final int rentalId;
  final int rating;
  final String? comment;
  final DateTime createdAt;
  final DateTime? updatedAt;

  ReviewModel({
    required this.id,
    required this.userId,
    this.userName,
    required this.toolId,
    this.toolName,
    required this.rentalId,
    required this.rating,
    this.comment,
    required this.createdAt,
    this.updatedAt,
  });

  factory ReviewModel.fromJson(Map<String, dynamic> json) {
    DateTime? parseDateTime(dynamic value) {
      if (value == null) return null;
      if (value is String) {
        try {
          return DateTime.parse(value);
        } catch (e) {
          return null;
        }
      }
      return null;
    }

    return ReviewModel(
      id: json['id'] as int,
      userId: json['userId'] as int,
      userName: json['userName'],
      toolId: json['toolId'] as int,
      toolName: json['toolName'],
      rentalId: json['rentalId'] as int,
      rating: json['rating'] as int,
      comment: json['comment'],
      createdAt: parseDateTime(json['createdAt']) ?? DateTime.now(),
      updatedAt: parseDateTime(json['updatedAt']),
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'userId': userId,
        'userName': userName,
        'toolId': toolId,
        'toolName': toolName,
        'rentalId': rentalId,
        'rating': rating,
        'comment': comment,
        'createdAt': createdAt.toIso8601String(),
        'updatedAt': updatedAt?.toIso8601String(),
      };
}

